File: examples/modulith-clinical-tool-jpa/src/main/java/io/zenwave360/example/clinicaltool/modules/masterdata/MasterDataRepository.java
package io.zenwave360.example.clinicaltool.modules.masterdata;

import io.zenwave360.example.clinicaltool.modules.masterdata.domain.*;
import java.util.*;
import org.springframework.data.jpa.repository.*;
import org.springframework.stereotype.Repository;

/**
 * Spring Data JPA repository for the MasterData entity.
 */
@SuppressWarnings("unused")
@Repository
public interface MasterDataRepository extends JpaRepository<MasterData, Long> {
    java.util.Optional<MasterData> findByTypeAndKey(MasterDataType type, String key);

    List<MasterData> findByType(MasterDataType type);
}

File: examples/modulith-klinical-tool-jpa/src/main/kotlin/io/zenwave360/example/clinicaltool/modules/masterdata/MasterDataRepository.kt
package io.zenwave360.example.clinicaltool.modules.masterdata

import io.zenwave360.example.clinicaltool.modules.masterdata.domain.*
import java.math.*
import java.time.*
import java.util.*
import org.springframework.data.jpa.repository.*
import org.springframework.stereotype.Repository

/** Spring Data JPA repository for the MasterData entity. */
@Suppress("unused")
@Repository
interface MasterDataRepository : JpaRepository<MasterData, Long> {
    fun findByTypeAndKey(type: MasterDataType, key: String): MasterData?
}
